ckan/migration/migrate_package_activity.py

```diff
@@ -67,15 +66,25 @@ def num_activities_migratable():
     from ckan import model
     num_activities = model.Session.execute(u'''
     SELECT count(*) FROM activity a JOIN package p ON a.object_id=p.id
-    WHERE a.activity_type IN ('new package', 'changed package')
-    AND p.private = false;
+    WHERE a.activity_type IN ('new package', 'changed package');
     ''').fetchall()[0][0]
     return num_activities
 
 
 def migrate_all_datasets():
     import ckan.logic as logic
-    dataset_names = logic.get_action(u'package_list')(get_context(), {})
+    dataset_names = []
+    offset = 0
+    count = 1
+    while len(dataset_names) < count:
+        dn = logic.get_action(u'package_search')(get_context(),
+                                                {'include_private': True,
+                                                'fl': ['id'],
+                                                'rows': 5000,
+                                                'start': offset})
+        offset += 5000
+        count = dn.get('count')
+        dataset_names += [result.get('id') for result in dn.get('results')]
```

Comment:
```suggestion
        dataset_names += [result['id'] for result in dn['results']]
```
better to get a `KeyError` than a list full of None values or a `TypeError: 'NoneType' object is not iterable` (and it's shorter too)